### PR TITLE
Property based tests on TxPool: pending, locals

### PR DESCRIPTION
**Pull request by JekaMas**

### CODE REVIEW

File: core/tx_pool_test.go [hunk 1]
```diff
@@ -2561,3 +2571,342 @@ func BenchmarkPoolMultiAccountBatchInsert(b *testing.B) {
...
+		// Generate a batch of transactions to enqueue into the pool
+		testTxPoolConfig := testTxPoolConfig
+
+		// from sentry config
+		testTxPoolConfig.AccountQueue = 16
+		testTxPoolConfig.AccountSlots = 16
+		testTxPoolConfig.GlobalQueue = 32768
+		testTxPoolConfig.GlobalSlots = 32768
+		testTxPoolConfig.Lifetime = time.Hour + 30*time.Minute //"1h30m0s"
+		testTxPoolConfig.PriceLimit = 30000000000
```
Comment (0xsharma) [inline core/tx_pool_test.go:2703]: Can we get these configs from params directly ?

Comment (JekaMas) [reply core/tx_pool_test.go:2703]: Do we have a predefined config with that exact params?

Comment (0xsharma) [reply core/tx_pool_test.go:2703]: We can try getting from [here](https://github.com/maticnetwork/bor/blob/9dcc9935663a87425162d2d367c83f6d0bded2a8/internal/cli/server/config.go#L387).

Comment (JekaMas) [reply core/tx_pool_test.go:2703]: Why we introduced so many copy-paste types in cli/server ? like TxPoolConfig?

Comment (0xsharma) [reply core/tx_pool_test.go:2703]: Mostly to get a generic config, I guess.

**Comment by JekaMas** [comment]
0txs blocks are reproducible as well as stuck in txPool transactions. For that reason the rest is marked as skipped for further txPool fixes.

File: core/tx_pool.go [hunk 2]
```diff
@@ -1833,3 +1852,9 @@ func (t *txLookup) RemotesBelowTip(threshold *big.Int) types.Transactions {
 func numSlots(tx *types.Transaction) int {
 	return int((tx.Size() + txSlotSize - 1) / txSlotSize)
 }
+
+func MakeWithPromoteTxCh(ch chan struct{}) func(*TxPool) {
```
Comment (temaniarpit27) [inline core/tx_pool.go:1856]: where are we calling this PR @JekaMas

Comment (temaniarpit27) [inline core/tx_pool.go:1856]: can we move this to test file if it called there only?

**Review by temaniarpit27 — APPROVED** [review]

**Review by marcello33 — APPROVED** [review]

**Review by temaniarpit27 — APPROVED** [review]

**Comment by temaniarpit27** [comment]
@JekaMas can you please check the CI?

**Pull request merged**